src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs

```diff
@@ -1051,9 +1051,20 @@ public async Task RequestToActiveConnectionIdKillsPreviousConnectionLongPolling(
                 Assert.True(request1.IsCompleted);
 
                 request1 = dispatcher.ExecuteAsync(context1, options, app);
+                // Wait until the request has started internally
+                while (connection.TransportTask.IsCompleted)
```

Comment:
I believe this test was flaky before (although no failures happened) it was just a very small race, but now that I added another statemachine the test became more flaky and this is my "solution" to make it consistent.

Before we dispatched 2 polls and assumed that one would start before the other, but they actually race each other once the first one goes async. I couldn't find a good place to inject our own sync points so am waiting on the `TransportTask` to change meaning the request has gotten far enough.

If someone has a better idea please help :)

Reply:
Might want to add a maximum wait here...